duckduckgo/Android
# Add email survey

Task/Issue URL:

### Description
This PR adds a few changes around surveys to be able to support a new email survey.

### Steps to test this PR

In the `SurveyService`, change the url to `http://ddg-sandbox.s3.amazonaws.com/survey/v2/survey-mobile.json`

_Survey doesn't show up in old app versions_
1. Build and launch the app from develop
1. In the app database check the survey table, you should the survey schedule with `daysInstalled = -1`
1. Perform a search and click `Hide` when the CTA shows up and then `Hide tips forever` to dismiss all the onboarding CTAs
1. Go to the new tab (home) screen, the survey CTA should not show up.

_Survey doesn't show up in old app versions after more than 30 days_
1. In the `AppInstallStore` change the method `AppInstallStore.daysInstalled()` to return 40L.
1. Rebuild and launch the app
1. Go to the home screen, the survey CTA should not show up.

_Survey doesn't show up if not signed in_
1. Revert the changes done in `AppInstallStore` and go back to this PR's branch. Do not uninstall the app.
1. Rebuild and launch the app
1. Go to the home screen, the survey CTA should not show up.
1. In the app database, the `survey` table should be empty

_Survey shows up if signed in_
1. Sign in to your email account.
1. Close the app and launch it again.
1. Go to the home screen, the survey CTA should show up.
1. In the app database, the `survey` table should contain one entry. The URL on that entry should include a query parameter called cohort with your email cohort.

_Survey closes when finished_
1. Take the survey
1. After finishing the survey you should be taken back to the home screen and the survey CTA should not show up.
